jax/core.py

```diff
@@ -1166,15 +1178,32 @@ def process_env_traces(primitive: Union['CallPrimitive', 'MapPrimitive'],
     trace = ans._trace.main.with_cur_sublevel()
     outs = map(trace.full_raise, outs)
     outs, cur_todo = primitive.post_process(trace, outs, params)
+    if isinstance(primitive, MapPrimitive):
```

Comment:
But then I would have to replicate the rest of the code of `process_env_traces` in another function. So I don't see how the conditional is wore than that?

Reply:
Yeah, I think it's simpler to have two separate, simpler functions with a few lines in common (process_env_traces isn't that long, and common parts can be abstracted into function calls) rather than to have conditional logic under one function. But it's not a strong feeling; I could be wrong, and we can always iterate later.